src/devices/LidarLiteV3/LidarLiteV3.cs

```diff
@@ -0,0 +1,363 @@
+            }
+
+            _gpioController?.Write(_powerEnablePin, PinValue.Low);
+        }
+
+        /// <summary>
+        /// Power on the device if GPIO controller and power enable pin is provided.
+        /// </summary>
+        public void PowerOn()
+        {
+            if (_powerEnablePin == -1)
+            {
+                throw new InvalidOperationException("Cannot power off without providing GPIO controller and power enable pin.");
+            }
+
+            _gpioController?.Write(_powerEnablePin, PinValue.High);
+        }
+
+        /// <summary>
+        /// Reset FPGA, all registers return to default values
+        /// </summary>
+        public void Reset()
+        {
+            try
+            {
+                WriteRegister(Register.ACQ_COMMAND, 0x00);
+            }
+            catch (IOException ex)
+            {
+                // `IOException: Error 121 performing I2C data transfer.` is thrown every time
+                // the device is reset on the Raspberry PI.  I think the reset signal causes
+                // a disruption in the I2C connection.  Without shallowing this exception,
+                // Reset() would always throw the IOException.
+                if (ex.Message != "Error 121 performing I2C data transfer.")
+                {
+                    throw;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Measure distance.
+        /// </summary>
+        /// <remarks>
+        /// Note: Do not call if running while in repetition mode.  It will block until
+        /// repetition finishes (forever if infinite).
+        /// </remarks>
+        /// <param name="withReceiverBiasCorrection">Faster without bias correction, but more prone to errors if condition changes.</param>
+        /// <returns>Distance as a length unit.</returns>
+        public Length MeasureDistance(bool withReceiverBiasCorrection = true)
+        {
+            WriteRegister(Register.ACQ_COMMAND, withReceiverBiasCorrection ? (byte)0x04 : (byte)0x03);
+
+            while (Status.HasFlag(SystemStatus.BusyFlag))
+            {
+                DelayHelper.DelayMicroseconds(4, allowThreadYield: true);
+            }
+
+            return LastDistance;
+        }
+
+        /// <summary>
+        /// Set the repetition mode to enable automatic measurement.
+        /// </summary>
+        /// <param name="measurementRepetition">Repetition mode, either Off, Repeat, or RepeatInfinitely.</param>
+        /// <param name="count">If Repeat, the number of times to repeat the measurement.</param>
+        /// <param name="delay">The delay between each measurements. Note the unit does not directly to hz, a value of 20 maps to about 100 hz.</param>
+        public void SetMeasurementRepetitionMode(MeasurementRepetition measurementRepetition, int count = -1, int delay = -1)
```

Comment:
ideally `delay` should be of type `TimeSpan?`

Reply:
There isn't a direct correlation between delay and real-world time.  Documentation only states that a value of 20 is about 100 hz, but 40 is not necessarily 200 hz.  Possible future enhancement is to build a mapping table, but I don't have equipment to accurately measure how often it does the measurement.